Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.ts

```diff
@@ -268,20 +305,177 @@ namespace gdjs {
+        this.updatePositionIfNeeded();
+        this._graphics.updateTransform();
+      }
+    }
+
+    updateAngle(): void {
+      if (this._object._useAbsoluteCoordinates) {
+        this._graphics.angle = 0;
+      } else {
+        this._graphics.angle = this._object.angle;
+      }
+      this._transformationIsUpToDate = false;
+    }
+
+    updateScaleX(): void {
+      if (this._object._useAbsoluteCoordinates) {
+        this._graphics.scale.x = 1;
+      } else {
+        this._graphics.scale.x = this._object._scaleX;
+      }
+      // updatePositionX() uses scale.x
+      this._positionXIsUpToDate = false;
+      this._transformationIsUpToDate = false;
+    }
+
+    updateScaleY(): void {
+      if (this._object._useAbsoluteCoordinates) {
+        this._graphics.scale.y = 1;
+      } else {
+        this._graphics.scale.y = this._object._scaleY;
+      }
+      // updatePositionY() uses scale.y
+      this._positionYIsUpToDate = false;
+      this._transformationIsUpToDate = false;
+    }
+
+    getDrawableX(): float {
+      if (this._object._useAbsoluteCoordinates) {
+        return this._graphics.getLocalBounds().left;
+      }
+      let localBound = this._graphics.getLocalBounds().left;
+      if (this._object._flippedX) {
+        const rotationCenterX = this._object.getRotationCenterX();
+        localBound = 2 * rotationCenterX - localBound;
+      }
+      // When new shape are drawn, the bounds of the object can extend.
+      // The object position stays the same but (drawableX; drawableY) can change.
+      return (
+        this._object.getX() + localBound * Math.abs(this._graphics.scale.x)
+      );
+    }
+
+    getDrawableY(): float {
+      if (this._object._useAbsoluteCoordinates) {
+        return this._graphics.getLocalBounds().top;
       }
+      let localBound = this._graphics.getLocalBounds().top;
+      if (this._object._flippedY) {
+        const rotationCenterY = this._object.getRotationCenterY();
+        localBound = 2 * rotationCenterY - localBound;
+      }
+      return (
+        this._object.getY() + localBound * Math.abs(this._graphics.scale.y)
+      );
+    }
+
+    getWidth(): float {
+      return this._graphics.width;
+    }
+
+    getHeight(): float {
+      return this._graphics.height;
+    }
+
+    getUnscaledWidth(): float {
+      return this._graphics.getLocalBounds().width;
+    }
+
+    getUnscaledHeight(): float {
+      return this._graphics.getLocalBounds().height;
+    }
+
+    /**
+     * @returns The drawing origin relatively to the drawable top left corner.
+     */
+    getOriginX() {
+      return -this._graphics.getLocalBounds().left;
+    }
+
+    /**
+     * @returns The drawing origin relatively to the drawable top left corner.
+     */
+    getOriginY() {
+      return -this._graphics.getLocalBounds().top;
+    }
+
+    transformToDrawing(point: FloatPoint): FloatPoint {
+      this.updateTransformationIfNeeded();
+      this._graphics.updateTransform();
```

Comment:
This seems useless, I'm removing this line.

Reply:
This is useful. Otherwise, there is a one frame delay. Try to change the angle by 20° and display the hitboxes.

Reply:
No, sorry, the call to this._graphics.updateTransform is done by updateTransformationIfNeeded. It can be removed.

Reply:
Thanks for checking 👍